src/app/tabs/newAdmin/subdomains/operations.js

```diff
@@ -43,33 +46,38 @@ const registerSubdomain = (
   parentDomain, subdomain, newOwner, setupResolution,
 ) => async (dispatch) => {
   dispatch(waitingNewSubdomainConfirm());
-  const transactionConfirmed = result => () => {
+
+  const signer = await getSigner();
+
+  const r = rns(signer);
+
+  try {
+    // eslint-disable-next-line no-debugger
+    debugger;
+
+    const result = await (await r.setSubdomainOwner(parentDomain, subdomain, newOwner)).wait();
+
+    if (setupResolution) {
+      await (await r.setResolver(`${subdomain}.${parentDomain}`, publicResolver)).wait();
+      const addrResolver = resolver(signer);
+      await (await addrResolver.setAddr(`${subdomain}.${parentDomain}`, newOwner)).wait();
+    }
+
     dispatch(addSubdomainToList(subdomain, newOwner));
-    dispatch(receiveNewSubdomain(result));
+    dispatch(receiveNewSubdomain(result.transactionHash));
     updateSubdomainToLocalStorage(parentDomain, subdomain, true);
     sendBrowserNotification(`${subdomain}.${parentDomain}`, 'register_subdomain');
-  };
-
-  const method = setupResolution
-    ? getRNS().subdomains.create(parentDomain, subdomain, newOwner, newOwner, { gas: 85000 })
-    : getRNS().subdomains.create(parentDomain, subdomain);
-
-  method
-    .then(result => dispatch(transactionListener(
-      result,
-      () => transactionConfirmed(result),
-      {},
-      errorReason => dispatch(errorNewSubdomain(errorReason.message)),
-    )))
-    .catch(error => dispatch(errorNewSubdomain(error.message)));
+  } catch (error) {
+    dispatch(errorNewSubdomain(error.message));
+  }
 };
 
 const getSubdomainOwner = (domain, subdomain) => async (dispatch) => {
   const hash = namehash(`${subdomain}.${domain}`);
 
-  const rns = getRNS();
-  await rns.compose();
-  await rns.contracts.registry.methods.owner(hash).call((error, result) => {
+  const r = getRNS();
+  await r.compose();
+  await r.contracts.registry.methods.owner(hash).call((error, result) => {
```

Comment:
this was not possible using the SDK?

Reply:
These are not part of the flow, will make the change in another issue.